cmd/agent/app/config.go

```diff
@@ -18,42 +20,59 @@ import (
 
 func init() {
 	AgentCmd.AddCommand(configCommand)
+	configCommand.AddCommand(listRuntimeCommand)
+	configCommand.AddCommand(setCommand)
 }
 
-var configCommand = &cobra.Command{
-	Use:   "config",
-	Short: "Print the runtime configuration of a running agent",
-	Long:  ``,
-	RunE: func(cmd *cobra.Command, args []string) error {
+var (
+	configCommand = &cobra.Command{
+		Use:   "config",
+		Short: "Print the runtime configuration of a running agent",
+		Long:  ``,
+		RunE:  showRuntimeConfiguration,
+	}
+	listRuntimeCommand = &cobra.Command{
+		Use:   "list-runtime",
+		Short: "List settings that can be changed at runtime",
+		Long:  ``,
+		RunE:  listRuntimeConfigurableValue,
+	}
+	setCommand = &cobra.Command{
```

Comment:
Maybe a `get <setting>` method could be useful, what do you think?

Reply:
I agree, but as per the the design discussion I believed that list-runtime was deemed to be enough, anyway it can be added relatively easily, but I'm not 100% sure it is very useful.

Reply:
I was thinking about the other Agents which are getting some info from the infra agent, calling it with some commands (for instance the hostname). There is a RFC in the pipe to do this better, but because a `get` is easy to implement that's probably worth (avoid some parsing logic)? We could discuss this with the team if you want to.

Reply:
I implemented a `get` method.